### PR TITLE
UploadV2 Backend

### PR DESCRIPTION
**Pull request by Rin-chan**

### CODE REVIEW

**Comment by Rin-chan** [comment]
[IgnoreAntiforgeryToken] Should be removed in DraftController. However I can't seem to run it with this removed.

**Comment by Rin-chan** [comment]
> [IgnoreAntiforgeryToken] Should be removed in DraftController. However I can't seem to run it with this removed.

Fixed

File: internet-webapp/MediaLibrary.Internet.Web/assets/scripts/UploadPortal/components/ImageInput.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+// Our app
+const FileInput = () => {
+  const formContext = useForm()
+
+  return (
+    <div className="App">
+      <FilePond
+        allowFileTypeValidation={true}
+        acceptedFileTypes={['image/png', 'image/jpg', 'image/jpeg', 'image/gif', 'image/bmp', 'image/heic']}
```
Comment (khteo) [inline internet-webapp/MediaLibrary.Internet.Web/assets/scripts/UploadPortal/components/ImageInput.js:31]: Suggest to retest the upload behaviour on mobile browsers now that we are no longer using an vanilla <input> tag. Does iOS still sanitise the HEIC files and re-compress to JPEG?

**Review by khteo — CHANGES_REQUESTED** [review]

File: internet-webapp/MediaLibrary.Internet.Web/Background/ScheduledService.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    /// <summary>
+    /// A recurring background job to delete Draft after some time
+    /// </summary>
+    public class ScheduledService : BackgroundService
+    {
+        private CrontabSchedule _schedule;
+        private DateTime _nextRun;
+
+        // Run once at every second minute
+        private static readonly string Schedule = "*/2 * * * *";
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/Background/ScheduledService.cs]: Changed to "59 23 * * *"

File: internet-webapp/MediaLibrary.Internet.Web/Controllers/DraftController.cs [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+            catch (Exception e)
+            {
+                _logger.LogError(e, "File {FileName} failed to upload.", untrustedFileName);
+
+                ModelState.AddModelError(file.Name, $"File {encodedFileName} could not be uploaded.");
+                TempData["Alert.Type"] = "danger";
+                TempData["Alert.Message"] = "Failed to upload files. Please correct the errors and try again.";
+                return;
+            }
+            return;
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/Controllers/DraftController.cs]: Removed

File: internet-webapp/MediaLibrary.Internet.Web/MediaLibrary.Internet.Web.csproj [hunk 4]
```diff
@@ -27,6 +27,10 @@
     <None Include="$(SpaRoot)**" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\MediaLibrary.Internet.Api\MediaLibrary.Internet.Api.csproj" />
+  </ItemGroup>
+
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/MediaLibrary.Internet.Web.csproj]: Removed

File: internet-webapp/MediaLibrary.Internet.Web/assets/scripts/UploadPortal/components/AllContext.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    Location: "Serangoon",
+    Copyright: "URA",
+    Project: "SERANGOON",
+    Caption: "a land with cranes",
+    Tag: "sand,land,crane",
+    DateTaken: "28th June 2022, 19:28 PM",
+    UploadDate: "28th June 2022, 19:28 PM",
+    AdditionalField: []
+    }
+  ])
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/assets/scripts/UploadPortal/components/AllContext.js]: Removed

File: internet-webapp/package.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/package.json]: Removed

File: internet-webapp/MediaLibrary.Internet.Web/assets/scripts/UploadPortal/components/PreviewUpdate.js [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+        headers: {
+          RequestVerificationToken: document.querySelector('meta[name="RequestVerificationToken"]').content
+        }
+      })
+    }
+
+    closeModal()
+    setTimeout(() => {
+      renderRefresh();
+    }, 1000)
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/assets/scripts/UploadPortal/components/PreviewUpdate.js]: Removed most of the setTimeout except for "setTimeout(() => { formContext.setAlertActive(false) }, 5000)" in Stepper.js as this displays the alert to tell user that the file has been uploaded into intranet.

File: internet-webapp/MediaLibrary.Internet.Web/Controllers/DraftController.cs [hunk 8]
```diff
@@ -0,0 +1,843 @@
...
+            sasBuilder.ExpiresOn = DateTimeOffset.UtcNow.AddHours(1); // Only valid for one hour.
+            sasBuilder.SetPermissions(BlobContainerSasPermissions.Read); // This SAS token can only be used for reading (prevent misuse to write/delete)
+
+            Uri sasUri = _blobContainerClient.GenerateSasUri(sasBuilder);
+
+            string[] uri = sasUri.ToString().Split("?");
+            string url = "?" + uri[1];
+
+            return url;
+        }
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/Controllers/DraftController.cs]: Done

File: internet-webapp/MediaLibrary.Internet.Web/AppSettings.cs [hunk 9]
```diff
@@ -4,6 +4,7 @@ public class AppSettings
     {
         public string MediaStorageConnectionString { get; set; }
         public string MediaStorageContainer { get; set; }
+        public string MediaStorageConnectionStringImage { get; set; }
```
[the remark this answers is not in the document]
Comment (Rin-chan) [reply internet-webapp/MediaLibrary.Internet.Web/AppSettings.cs:7]: MediaStorageConnectionStringImage uses the same string from MediaStorageConnectionString in Intranet

**Review by khteo — APPROVED** [review]
+1

**Pull request merged**